dist/sessions.js

```diff
@@ -43,6 +59,87 @@ class Sessions {
       };
     });
   }
+  /** Parse a JWT and verify the signature, preferring local verification over remote.
+   *
+   * If maxTokenAge is set, remote verification will be forced if the JWT was issued at (based on
+   * the "iat" claim) more than maxTokenAge seconds ago.
+   */
+
+
+  async authenticate_jwt(jwt, options) {
+    try {
+      const session = await this.authenticate_jwt_local(jwt, options);
+      return {
+        session,
+        session_jwt: jwt,
+        session_token: "",
+        // Pretend that this was a successful API request.
+        request_id: "",
+        status_code: 200
+      };
+    } catch (err) {
+      if (err instanceof _errors.ClientError && err.code === "jwt_too_old") {
+        // Token was too old (stale) to verify locally. Check with the Stytch API.
+        return this.authenticate({
+          session_jwt: jwt
+        });
+      }
+
+      throw err;
+    }
+  }
+  /** Parse a JWT and verify the signature locally (without making an /authenticate call).
+   *
+   * If maxTokenAge is set, this will return an error if the JWT was issued (based on the "iat"
+   * claim) more than maxTokenAge seconds ago.
+   */
+
+
+  async authenticate_jwt_local(jwt, options) {
+    // Don't pass maxTokenAge directly to jwtVerify because it interprets zero as "infinity". We
+    // want zero to mean "every token is stale" and force remote verification.
+    const maxTokenAge = options === null || options === void 0 ? void 0 : options.maxTokenAge;
+    const now = Date.now() / 1000; // Unix epoch seconds
+
+    let payload;
+
+    try {
+      const token = await jose.jwtVerify(jwt, this.jwks, this.jwtOptions);
+      payload = token.payload;
+    } catch (err) {
+      throw new _errors.ClientError("jwt_invalid", "Could not verify JWT", err);
+    }
+
+    if (maxTokenAge != null) {
+      const iat = payload.iat;
+
+      if (!iat) {
+        throw new _errors.ClientError("jwt_invalid", "JWT was missing iat claim");
+      }
+
+      if (now - iat > maxTokenAge) {
+        throw new _errors.ClientError("jwt_too_old", `JWT was issued at ${iat}, more than ${maxTokenAge} seconds ago`);
+      }
+    } // Re-pack the JWT contents as session data. Since we're actually changing the types of
```

Comment:
New line for this comment here.

Reply:
This one's in the compiled output, so I'm going to skip it. If you feel strongly about it, my browser history tells me I've looked into this before lol: https://github.com/babel/babel/issues/8437#issuecomment-623853285

Reply:
The whole comment went away, so I get to resolve this for free 😛